### PR TITLE
[Notifier] Improve the error on email channel sender

### PR DESCRIPTION
**Pull request by plandolt**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      |no
| New feature?  |no
| Deprecations? |no 
| License       | MIT

It took me too much time to figure out its not the header from but the envelope sender.

### CODE REVIEW

File: src/Symfony/Component/Notifier/Channel/EmailChannel.php [hunk 1]
```diff
@@ -57,7 +57,7 @@ public function notify(Notification $notification, RecipientInterface $recipient
         if ($email instanceof Email) {
             if (!$email->getFrom()) {
                 if (null === $this->from) {
-                    throw new LogicException(sprintf('To send the "%s" notification by email, you should either configure a global "from" or set it in the "asEmailMessage()" method.', get_debug_type($notification)));
+                    throw new LogicException(sprintf('To send the "%s" notification by email, you should either configure a global envelope sender or set it in the "asEmailMessage()" method.', get_debug_type($notification)));
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Channel/EmailChannel.php]: Hmm before it was more clear to me, maybe we should not improve the message that way, but keen the `"from"` and add more information where one can do this?

cc @jschaedl

Comment (jschaedl) [reply src/Symfony/Component/Notifier/Channel/EmailChannel.php]: The envelope sender is only a fallback:
https://github.com/symfony/symfony/blob/e7ed342ab9d3dfd33510e8f93f17929d3548ee7f/src/Symfony/Component/Notifier/Channel/EmailChannel.php#L44

What about:

```suggestion
                    throw new LogicException(sprintf('To send the "%s" notification by email, you should either configure a global "from", set a sender in the global "envelope" or set a "from" in the "asEmailMessage()" method.', get_debug_type($notification)));
```

Comment (plandolt) [reply src/Symfony/Component/Notifier/Channel/EmailChannel.php]: Yes its a fallback on L44 of this service but the default way the framework extension is setting the from argument in https://github.com/symfony/symfony/blob/e7ed342ab9d3dfd33510e8f93f17929d3548ee7f/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L2394:L2399.

What about this (a bit more verbose)?
```suggestion
                    throw new LogicException(sprintf('To send the "%s" notification by email, you should either configure a global "from", set a sender in the global "envelope" of the mailer configuration or set a "from" in the "asEmailMessage()" method.', get_debug_type($notification)));
```

Comment (jschaedl) [reply src/Symfony/Component/Notifier/Channel/EmailChannel.php]: LGTM

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
@nicolas-grekas open for a `5.2` merge?

File: src/Symfony/Component/Notifier/Channel/EmailChannel.php [hunk 2]
```diff
@@ -57,7 +57,7 @@ public function notify(Notification $notification, RecipientInterface $recipient
         if ($email instanceof Email) {
             if (!$email->getFrom()) {
                 if (null === $this->from) {
-                    throw new LogicException(sprintf('To send the "%s" notification by email, you should either configure a global "from" or set it in the "asEmailMessage()" method.', get_debug_type($notification)));
+                    throw new LogicException(sprintf('To send the "%s" notification by email, you should either configure a global "from", set a sender in the global "envelope" of the mailer configuration or set a "from" in the "asEmailMessage()" method.', get_debug_type($notification)));
```
Comment (fabpot) [inline src/Symfony/Component/Notifier/Channel/EmailChannel.php]: a "from" header?

**Comment by fabpot** [comment]
We don't change error messages in patch releases, except for obvious mistakes.

**Comment by OskarStark** [comment]
open for finishing this PR?

**Comment by plandolt** [comment]
> open for finishing this PR?

Yes. I missed the mail notification of fabpot's review.

File: src/Symfony/Component/Notifier/Channel/EmailChannel.php (hunk 2 above)
Comment (plandolt) [reply src/Symfony/Component/Notifier/Channel/EmailChannel.php]: I added "header" to both "from"

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @scuben.

**Pull request merged**